src/lib.rs

```diff
@@ -612,8 +612,12 @@ impl<T> SliceDeque<T> {
     #[inline]
     pub fn reserve(&mut self, additional: usize) {
         let old_len = self.len();
-        let new_cap = old_len.checked_add(additional).expect("overflow");
-        self.reserve_capacity(new_cap);
+        let req_cap = old_len.checked_add(additional).expect("overflow");
+        let cur_cap = self.capacity();
+        if req_cap > cur_cap {
+            let dbl_cap = cur_cap * 2;
```

Comment:
This should probably be a `checked_mul`.

Reply:
I've turned this into a saturating multiply since this will give well defined behaviour and the will cause the allocation later to fail as expected.